manager/dispatcher/dispatcher_test.go

```diff
@@ -634,15 +724,31 @@ func TestAssignmentsAddingTasks(t *testing.T) {
 	assert.Empty(t, resp.Changes)
 
 	// create the relevant secrets, configs, and tasks and update the tasks
-	secrets, configs, tasks := makeTasksAndDependencies(t, nodeID)
-	createdSecrets, createdConfigs := secrets[:len(secrets)-1], configs[:len(configs)-1]
+	secrets, configs, resourceRefs, tasks := genTasks(t, nodeID)
+	var createdSecrets []*api.Secret
+	var createdConfigs []*api.Config
+	if len(secrets) > 0 {
+		createdSecrets = secrets[:len(secrets)-1]
+	}
+	if len(configs) > 0 {
+		createdConfigs = configs[:len(configs)-1]
+	}
 	err = gd.Store.Update(func(tx store.Tx) error {
 		for _, secret := range createdSecrets {
-			assert.NoError(t, store.CreateSecret(tx, secret))
+			if store.GetSecret(tx, secret.GetID()) == nil {
```

Comment:
nit: Just use `secret.ID`. `GetID` exists for type-agnostic code that uses these through an interface.